### PR TITLE
Feature/add button labels and renderChallenges

### PR DESCRIPTION
**Pull request by El-Duckster**
This pull request introduces multiple updates and enhancements related to the Challenge Card component, booking functionality, and UI styling. These changes aim to improve usability, maintainability, and responsiveness.

changes:
1-Add Type Icon: Added a type icon to the Challenge Card to visually distinguish between different challenge types .
2- Add Booking Button: Implemented a booking button on the Challenge Card & Buttons dynamically adjust text based on challenge type.
3- Add Labels : Added labels to the Challenge Card to display associated tags for each challenge & render dynamically based on the labels property.
4- Implement Challenge Card Class and added Booking and rendering logic as a utikity function to be used in Display All Challenges, show top 3 challenges and filter.js
5- Update Card Styling: Updated the Challenge Card layout to ensure consistent  with the wireframe.

### CODE REVIEW

**Review by Fagerdahl — APPROVED** [review]
You are doing an amazing job with this Elena!

File: js/utilities/renderChallenges.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import Challenge from '../challengeClass.js';
+
+export const renderChallenges = (challenges, container, withButtons = true) => {
```
Comment (Kirill9m) [inline js/utilities/renderChallenges.js:3]: That is very smart!
Anything that repeats more than twice should be made into a function :)

File: js/challengeClass.js [hunk 2]
```diff
@@ -46,21 +47,22 @@ class Challenge {
     return challengeRating;
   }
 
-  mainPageBtn() {
-    const link = document.createElement("a");
-    link.className = "challenge__btn";
+  createBookChallengeBtn() {
```
Comment (Kirill9m) [inline js/challengeClass.js:50]: I agree,
more descriptive and clear!

**Comment by El-Duckster** [comment]
> Great job, Elena! The code has become more compact and structured and our common code sections have been merged! Let's do the pull request! P.S I've already found a couple of bugs, but they aren't related to this part of the code.

Thank you very much for your review. Perfect! can you please update the project board with the bugs that you have discovered so we could look at them and see if we need to fix them now or if we shall take them in  our upcoming course :)

**Pull request merged**